src/components/Button/Button.tsx

```diff
@@ -0,0 +1,16 @@
+import { OverridableComponent } from "@material-ui/core/OverridableComponent";
+import { Button as MacawButton, ButtonTypeMap } from "@saleor/macaw-ui";
+import { isExternalURL } from "@saleor/utils/urls";
+import React from "react";
+import { Link } from "react-router-dom";
+
+const _Button: React.FC<any> = React.forwardRef(({ href, ...props }, ref) => {
+  if (href && !isExternalURL(href)) {
+    return <MacawButton {...props} to={href} component={Link} ref={ref} />;
+  }
+
+  return <MacawButton href={href} {...props} ref={ref} />;
+});
+_Button.displayName = "Button";
+
+export const Button = _Button as OverridableComponent<ButtonTypeMap>;
```

Comment:
What's the reason for having 2 constants here (_Button and Button)? You can do the type assertion directly on the first one, if it's a readability thing and was intended then I'd also change it in SortableTableRow.tsx

Reply:
Yes, that was my initial implementation, however this doesn't work with TypeScript:
<img width="778" alt="CleanShot 2022-05-04 at 16 24 13@2x" src="https://user-images.githubusercontent.com/4144459/166703396-6cf1549d-60c1-407e-ae16-3b7a9b65bf06.png">
<img width="518" alt="CleanShot 2022-05-04 at 16 24 05@2x" src="https://user-images.githubusercontent.com/4144459/166703403-ee1199f1-2653-4054-a351-0acfd2dc11a7.png">

If we make the assertion in the same constant, then TypeScript infers the type from the component declaration, and it doesn't work nicely with the `OverridableComponent` from Material-UI. We can't use `as unknown as OverridableComponent` as well because this on the other hand breaks types from `React.fowardRef`.

However, if we use this hack, then TypeScript infers the type from our declaration `React.FC<any>` and it allows us to cast this type.